yarn-project/cli/src/index.ts

```diff
@@ -278,21 +283,52 @@ export function getProgram(log: LogFn, debugLogger: DebugLogger): Command {
   program
     .command('get-logs')
     .description('Gets all the unencrypted logs from L2 blocks in the range specified.')
-    .option('-f, --from <blockNum>', 'Initial block number for getting logs (defaults to 1).')
-    .option('-l, --limit <blockCount>', 'How many blocks to fetch (defaults to 100).')
+    .option('-tx, --txHash <txHash>', 'A transaction hash to get the receipt for.', parseOptionalTxHash)
+    .option(
+      '-f, --fromBlock <blockNum>',
+      'Initial block number for getting logs (defaults to 1).',
+      parseOptionalInteger,
+    )
+    .option('-t, --toBlock <blockNum>', 'Up to which block to fetch logs (defaults to latest).', parseOptionalInteger)
+    .option('-c, --contractAddress <contractAddress>', 'Contract address to filter logs by.', parseOptionalAztecAddress)
+    .option('-s, --selector <selector>', 'Event selector to filter logs by.', parseOptionalSelector)
     .addOption(pxeOption)
-    .action(async options => {
-      const { from, limit } = options;
-      const fromBlock = from ? parseInt(from) : 1;
-      const limitCount = limit ? parseInt(limit) : 100;
+    .option('--follow', 'If set, will keep polling for new logs until interrupted.')
+    .action(async ({ txHash, fromBlock, toBlock, contractAddress, selector, rpcUrl, follow }) => {
+      const client = await createCompatibleClient(rpcUrl, debugLogger);
 
-      const client = await createCompatibleClient(options.rpcUrl, debugLogger);
-      const logs = await client.getUnencryptedLogs(fromBlock, limitCount);
-      if (!logs.length) {
-        log(`No logs found in blocks ${fromBlock} to ${fromBlock + limitCount}`);
+      if (follow) {
+        if (txHash) throw Error('Cannot use --follow with --txHash');
+        if (toBlock) throw Error('Cannot use --follow with --toBlock');
+      }
+
+      const filter: LogFilter = { txHash, fromBlock, toBlock, contractAddress, selector };
+
+      const fetchLogs = async () => {
+        const logs = await client.getUnencryptedLogs(filter);
+
+        if (!logs.length) {
+          const filterOptions = Object.entries(filter)
+            .filter(([, value]) => value !== undefined)
+            .map(([key, value]) => `${key}: ${value}`)
+            .join(', ');
+          if (!follow) log(`No logs found for filter: {${filterOptions}}`);
+        } else {
+          if (!follow) log('Logs found: \n');
+          logs.forEach(unencryptedLog => log(unencryptedLog.toHumanReadable()));
+          // Set `fromBlock` to the block number of the next block after the block of last log we fetched.
+          filter.fromBlock = logs[logs.length - 1].blockNumber + 1;
+        }
+      };
```

Comment:
How should the CLI deal with `maxLogs` being hit? If I'm not using `follow`, should it make another request to the node to retrieve the next page automatically?

And if I'm using follow, what happens if the logs "page" breaks in the middle of a block? The `filter.fromBlock = logs[logs.length - 1].blockNumber + 1` line would cause some logs from the last block to be lost. And if we remove the `-1`, we'd show duplicated logs. Do we need another filter option like `from-index-within-block`?

Reply:
This is a bit tricky. I think the best approach is to modify the API to return maxLogsHit flag and make the API not include any logs from a block "in the middle of which" we've finished.

With this change the CLI would work immediately with the `--follow` flag. As for the non-follow mode I could show a warning with the next filter they should use.

There is a danger in case a block would contain more logs than maxLogsCount param but I can just put there an error if this happens. Does this all sound good to you?

Reply:
> maxLogsHit flag and make the API not include any logs from a block "in the middle of which" we've finished.

Removing logs from the result feels a bit awkward. I'd take inspiration from other paginated APIs that let you include the key of the last item returned to start from there (eg [DynamoDB](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Query.Pagination.html)). So we could include a parameter `--after-log` or `--exclusive-start-log` that accepts an identifier for the last log returned (eg `blockNum.txHash.logIndex`), and use that for the follow-up request. WDYT?

Reply:
@spalladino Good idea. That is the most elegant. I think it would be nice to keep the API return type as `ExtendedUnencryptedL2Log[]` so I would include the logId inside the `ExtendedUnencryptedL2Log` type instead. I think it's nice to have a globally unique log id.

Lmk if you agree.

Reply:
I'm on the fence between making the `logId` part of the payload returned, vs making it a function available in aztec.js that computes it from the log data.

But given that we're returning custom classes for most calls (ExtendedUnencryptedL2Log in this case), maybe we can just add it as a `getLogId` method that uses the available log info in the instance to compute it?

Reply:
Great idea having this only in a client. Will do that.

Reply:
@spalladino this is addressed now. I did a refactor and now the ExtendedUnencryptedLog returns a LogId which contains all the necessary info.

This is the only way I could think of which does not require an additional call to pixie.

LMK if it looks good to you.